flake.nix

```diff
@@ -26,7 +26,11 @@
       # upstream ideally) so this interface might be brittle, but avoids
       # accidentally passing a detached nixpkgs from its flake (or its follows)
       # on to consumers.
-      craneLib = crane.mkLib final;
+      craneLib = (crane.mkLib final).overrideScope (_: scopePrev: {
+        mkCargoDerivation = args: scopePrev.mkCargoDerivation ({
+          RUSTFLAGS = "-D warnings";
+        } // args);
+      });
```

Comment:
I'd prefer if this was done elsewhere, as it's rather aggressive to deny warnings (such as e.g. unused variables not being prefixed with underscores).

I think if the goal is to increase correctness of the codebase, adding a clippy flake check, and then putting it in CI makes much more sense, there are examples of such checks in the crane repo in the examples folder that could essentially be copied.

That way we don't break fast iteration speed of "compiler driven development" but instead move the validation of pedantics to the finalization of a contribution.

Reply:
Probably just me that primarily uses cargo (rather than `nix build`) for compiler-driven-dev (shrug). I'm fine with moving it to a flake check as long as it fails the GH pipeline protecting main :) Will look at it as soon as possible.